### PR TITLE
Add Type checking on websocket debugger client

### PR DESCRIPTION
**Pull request by arthuro555**
Adds type checks to websocketDebuggerClient.
Also contains a new script in GDJS package.json to generate doc.

### CODE REVIEW

**Comment by arthuro555** [comment]
Ready for review @4ian

File: GDJS/Runtime/websocket-debugger-client/websocket-debugger-client.js [hunk 1]
```diff
@@ -384,6 +381,7 @@ gdjs.WebsocketDebuggerClient.prototype._depthLimitedSerializer = function(
       if (maxDepth != null && thisPos > maxDepth) {
         return '[Max depth reached]';
       } else if (~stack.indexOf(value))
+      // @ts-ignore
```
Comment (arthuro555) [inline GDJS/Runtime/websocket-debugger-client/websocket-debugger-client.js:384]: That is ignored as it says "cycleRepeater might be undefined", but as you can see above it is checked for undefined and null and if it is one of both it get defined.

File: GDJS/Runtime/websocket-debugger-client/websocket-debugger-client.js [hunk 2]
```diff
@@ -347,6 +343,7 @@ gdjs.WebsocketDebuggerClient.prototype._circularSafeStringify = function(
 ) {
   return JSON.stringify(
     obj,
+      // @ts-ignore
```
Comment (arthuro555) [inline GDJS/Runtime/websocket-debugger-client/websocket-debugger-client.js:346]: That one is weir as it says the arguments are from the wrong type, but the types correspond. To be more precise it says that basically `DebuggerClientCycleReplacer !== DebuggerClientCycleReplacer` wich isn't true.

**Comment by arthuro555** [comment]
I commented the 2 ts-ignore I had to use and why

**Comment by arthuro555** [comment]
Semaphore uses an outdated version of node :eyes:

File: GDJS/Runtime/websocket-debugger-client/websocket-debugger-client.js (hunk 2 above)
Comment (4ian) [reply GDJS/Runtime/websocket-debugger-client/websocket-debugger-client.js:346]: Weird, no big deal then.

File: GDJS/docs/jsdoc.conf.json [hunk 3]
```diff
@@ -50,6 +50,7 @@
     "recurse": true,
     "private": false,
     "lenient": true,
+	"template": "docs/jaguarjs-jsdoc",
```
Comment (4ian) [inline GDJS/docs/jsdoc.conf.json:53]: Can you use spaces rather than tabs.

File: GDJS/package.json [hunk 4]
```diff
@@ -7,13 +7,15 @@
...
     "jsdoc-plugin-intersection": "^1.0.2",
     "typescript": "3.6.4"
   },
   "dependencies": {
     "jsdoc-autoprivate": "0.0.1"
   },
   "scripts": {
-    "check-types": "tsc"
+    "check-types": "tsc",
+	"generate-doc": "jsdoc -c docs/jsdoc.conf.json docs/DocMainPage.md"
```
Comment (4ian) [inline GDJS/package.json:19]: nitpicking: same, can you use spaces rather than tabs?

**Comment by 4ian** [comment]
I've updated the semaphore Node version. The two ts-ignore seems fair, there might be a way to avoid them, but no big deal!

**Comment by 4ian** [comment]
Looks good, thanks! 👍

**Pull request merged**